### PR TITLE
Fix 2D/RZ: ParticleHistogram

### PR DESCRIPTION
**Pull request by ax3l**
Fix out-of-bounds access in particle histogram in non-3D.

### CODE REVIEW

File: Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp [hunk 1]
```diff
@@ -152,8 +152,13 @@ void ParticleHistogram::ComputeDiags (int step)
         {
             auto const w  = p.rdata(PIdx::w);
             auto const x  = p.pos(0);
+#if AMREX_SPACEDIM == 3
```
Comment (ax3l) [inline Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp]: We can probably use `GetParticlePosition` here, I am just unsure what happens downstream in `fun_partparser` with the signaling `NaN` that will be in `y`.

Comment (atmyers) [reply Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp]: I think way to handle RZ here would be to use the `GetParticlePosition` functor, but unpack the result here and set `y` to `0` before handing to the `fun_partparser`.

Comment (ax3l) [reply Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp]: I think for RZ we would keep all three reconstructed cartesian values, just for 2D we would reset `y` to zero.
Although we need to test if `y` is really unconditionally accessed in `fun_partparser`.

**Review by atmyers — APPROVED** [review]

**Review by Yin-YinjianZhao — APPROVED** [review]
Thanks for the PR!

**Pull request merged**